### PR TITLE
Update configurable views and app JSON configurations

### LINKED ISSUE #1761: [Phase 1] UI/UX and functionality updates for the configuration views.
**Context**
- After the review of the configuration work we have identified a couple of places where we need to fit bugs or tweak the UI to fit the Design team's recommendation. 
- All the changes required are added to the issue here. 

**Implementation**
- [x] Update the Service Button view to the look and feel to fit the attached screenshot. Also, update the functionality to meet the explanation in the screenshot.  
     - [x] Service button greyed out without a border -- This is when the status is `UP COMING`
         - [ ] The upcoming status disable the button. If the button is clicked while in this state a SnackBar with a user-configured message should be shown. 
     - [x] Service button with a blue border and blue text -- This is when the button status is `DUE`
     - [x] Service button with red background and white text -- This is when the service button status is `OVER DUE` 
Screenshot 2022-10-19 at 14 57 35

- [x] Add a `enable` attribute for the fab button. If the fab button `enable = false` then the button should have a look and feel as the image attached below. This should also apply to the extended fab actions. 
    - [ ] If the disabled fab is clicked then a SnackBar with a user-configured message should be shown. 
diabled-fab

- [x] Add the ability to add icons to the fab button. The image below shows the position of the icon. This should behaviour show be the same for the extended fab actions. 
- [x] Adding the `weight` attribute the to row view definition.  This will help correctly adapt the UI when have to disable different UI components.  
- [x] Add a vertical spacer view for the views in a ROW view.  
- [x] Add a border-radius attribute to the different views we have i.e ROW, COLUMN, CARD, SERVICE CARD, SERVICE BUTTON, COMPOUND_TEXT e.t.c.  The image below will help with the three attribute addition requirements. 
weight to row

**Acceptance Criteria**
- [ ] The service button should have the described UI/UX
- [ ] It should be possible to set the enable attribute on the fab button and the extended actions. 
- [ ] It should be possible to add icons to the fab button and extended actions.  
- [ ] The ROW view should have a weight attribute. 
- [ ] We should have a vertical spacer view.
- [ ] We should be able to set the border radius to the relevant fhircore configuration views. 

Continuation: https://github.com/opensrp/fhircore/issues/1788

### LINKED ISSUE #1788: [Phase 2] UI/UX and functionality updates for the configuration views.
- [x] Add a register cache refresh action on the questionnaire onClose action. 
- [x] Add the ability to show SnackBars to FHIR Core. These are potential replacements for the Toasters
    - [x] Make the SnackBar dismiss time configurable. 
snack-bar-sample

- [x]  The `menuActionButton` on the Navigation should be hidden when the `visible` attribute is set to `false`.

Screenshot from 2022-10-29 16-03-09

- [x] Add the ability to configure different report configurations for different configuration menus. The screenshot below shows a situation where we can have 2 reports menu items in the same navigation drawer. 

Screenshot from 2022-10-29 16-07-57

- [x] Add an action attributes to the `COMPOUND_TEXT` view. 
    - [x] This should be able to open a register and profile.
    - [ ] Count the number of results in the register the action is to open. This should similar to the count used on the Navigation configuration. 

Screenshot from 2022-10-29 16-21-30

- [x] Add the ability to define the Register title (used on the top) on the register configs. Right the registers use the navigation item `display` item for display. We can make the `display` attribute on the register the default title. This way the `COMPOUND_TEXT` on click loading a register doesn't have to care about setting. 
     - [ ] Add the way to pass extra data to add the Register title. e.g when a CHW is viewing a household created by a specific CHA we would want to show the CHA's name on the Register title. 

Screenshot from 2022-10-29 16-22-50

- [x] Add an `enable` attribute to all the views we have on the FHIR core app. When a view is set as disabled the following should happen. 
    - [x] By default the view is not clickable. However, we can configure the onClick behavior. Some disabled views might be clickable. e.g On the Family profile a deceased family member is shown as disabled but we should be able to click on the row and launch their profile.

### PR DESCRIPTION
**Pull request by ellykits**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #1761 
Fixes #1788 

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by ellykits** [comment]
Refactored the `ViewRenderer` in this PR to fix the following and add these extra features:
1. Fixed rendering views on ROWs and COLUMNs (Rows and Columns were not rendering) 
2. Include `weight` attribute to assist with positioning of views in parent layout. The view size will be generated as a percentage of the the available screen size. 
3. Split `ViewGroupProperties` into `ColumnProperties` and `RowProperties` the both have different attributes 
4. Fixed ActionableButton  background color
5. Fixed ServiceButton border color 
6. Fixed icons not showing on Fabs and Drawer menu Item (Other clients option)
7. Add border radius to all Properties as well as other common attributes
8. Introduce a disable property for views to mark them as disabled

File: android/quest/build.gradle [hunk 1]
```diff
@@ -138,7 +138,8 @@ android {
             applicationIdSuffix ".ecbis"
             versionNameSuffix "-ecbis"
             versionCode 4
-            versionName "0.1.5-rc1"
+            versionName "0.1.6-rc1"
+            buildConfigField "String[]", "SYNC_STRATEGIES", '{"Location", "Organization"}'
```
Comment (FikriMilano) [inline android/quest/build.gradle:142]: Can we move this to the ApplicationConfig?
For reference https://github.com/opensrp/fhircore/issues/1778

File: android/quest/src/main/assets/configs/app/sync_config.json [hunk 2]
```diff
@@ -17,13 +17,11 @@
...
-        "name": "organization",
-        "code": "managing-entity.patient.organization",
+        "name": "_id",
+        "code": "_id",
         "base": [
           "Group"
-        ],
-        "type": "token",
-        "expression": "#organization"
+        ]
```
Comment (FikriMilano) [inline android/quest/src/main/assets/configs/app/sync_config.json:24]: Is this intentional? and what's the reason for this change?

**Review by FikriMilano — CHANGES_REQUESTED** [review]

File: android/quest/src/main/assets/configs/app/sync_config.json (hunk 2 above)
Comment (ellykits) [reply android/quest/src/main/assets/configs/app/sync_config.json:24]: Yea this was done intentionally (as a workaround) so as to sync all the Group resources including those used in stock.

**Review by pld — APPROVED** [review]

**Comment by pld** [comment]
@FikriMilano can you check to see if the changes you requested have been addressed? thank you!

**Comment by dubdabasoduba** [comment]
@ellykits What is remaining here? I would like to merge this to `main` then we can create issues for the tests and minor bugs. 

@pld @f-odhiambo
